dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/ObjectCreationNode.java

```diff
@@ -37,25 +57,62 @@ public ObjectCreationNode(
         this.constructor = constructor;
         this.arguments = arguments;
         this.classbody = classbody;
+        this.receiver = receiver;
     }
 
+    /**
+     * Returns the constructor node
+     *
+     * @return the constructor node
+     */
+    @Pure
     public Node getConstructor() {
         return constructor;
     }
 
+    /**
+     * Returns the arguments
+     *
+     * @return the arguments
+     */
+    @Pure
     public List<Node> getArguments() {
         return arguments;
     }
 
+    /**
+     * Returns the argument
+     *
+     * @param i the index of the argument
+     * @return the argument
+     */
+    @Pure
     public Node getArgument(int i) {
         return arguments.get(i);
     }
 
+    /**
+     * Returns the enclosing type receiver if exists
+     *
+     * @return the enclosing type receiver
+     */
+    @Pure
+    public @Nullable Node getReceiver() {
```

Comment:
Is this non-null only when the receiver was explicitly provided?
What happens when there is an enclosing receiver, but it is omitted in the source program?

Reply:
Thanks! I agree, so does Nullable here make sense?